### PR TITLE
dev: Allow easier overriding of views in developer applications.

### LINKED ISSUE #555: Dev: Extend Actions to allow custom rendering methods.
Currently, the `Email2FA` and `EmailActivation` methods use the built-in `view()` method to display a view. This works great unless someone has provided an extended View layer. For example, in Bonfire we have a theme system that uses `$this->render()` in the controllers to integrate with the theme system. This would require duplicating the actions into the developers code when all they want to do is change the render method. That leads to maintenance headaches. 

I propose we add a `renderView()` method to both of the Actions so that all a developer has to do is extend the Action class and override that one method, keeping all of the other logic intact.

### PR DESCRIPTION
**Pull request by lonnieezell**
Fixes  #555

When using a theme/template system other than the built in CI features, it is very difficult to integrate without copying large amounts of logic from Actions and controllers that a developer should not have to maintain. 

This adds a new trait, `Viewable`, that provides a simple, consistent way to override just how views are rendered.

### CODE REVIEW

File: docs/customization.md [hunk 1]
```diff
@@ -90,6 +91,25 @@ class LoginController extends ShieldLogin
...
+
+```php
+use Acme\Themes\Traits\Themeable;
+use CodeIgniter\Shield\Controllers\LoginController;
+
+class MyLoginController extends LoginController
+{
+    use Themable;
+
+    protected function view(string $view, array $data=[], array $options=[])
```
Comment (kenjis) [inline docs/customization.md:106]:
```suggestion
    protected function renderView(string $view, array $data = [], array $options = [])
```

**Review by MGatner — COMMENTED** [review]
Looks good to me with @kenjis' fix.

File: docs/customization.md [hunk 2]
```diff
@@ -90,6 +91,25 @@ class LoginController extends ShieldLogin
...
```
Comment (kenjis) [inline docs/customization.md:106]:
```suggestion
    protected function view(string $view, array $data = [], array $options = []): string
```

Comment (MGatner) [reply docs/customization.md:106]: Let's launch.

**Review by MGatner — APPROVED** [review]
@kenjis Last check?

**Review by kenjis — APPROVED** [review]

**Pull request merged**